src/engine/Server.h

```diff
@@ -77,6 +76,9 @@ class Server {
 
   mutable net::static_thread_pool threadPool_;
 
+  /// Executor with a single thread that is used to run timers asynchronously.
+  mutable net::static_thread_pool timerExecutor_{1};
```

Comment:
I think that those shouldn't be mutable, I would argue that the calling functions are simply not const then.